modules/imgcodecs/src/loadsave.cpp

```diff
@@ -83,6 +83,9 @@ static Size validateInputImageSize(const Size& size)
 
 static inline int calcType(int type, int flags)
 {
+    if (flags == IMREAD_ADD_ALPHA)
```

Comment:
There is [cv::IMREAD_ANYCOLOR](https://docs.opencv.org/4.x/d8/d6a/group__imgcodecs__flags.html#gga61d9b0126a3e57d9277ac48327799c80ab6573b69300c092b61800222fe555953) option. It should presume alpha channel, if it's present in the original image.

Reply:
see the issue #21695 related cv::IMREAD_ANYCOLOR